### PR TITLE
Refactor base toolbox code de-coupled from app & config into tool_util.

### PR DESCRIPTION
**Pull request by jmchilton**
This was originally decoupled by me for this purpose and this code was in galaxy-lib years ago, but it just never made it into galaxy-tool-util. This commit clears up that oversight I think.

## How to test the changes?
- [x] This is a refactoring of components with existing test coverage.

## License
- [x] I agree to license these contributions under [Galaxy's current license](https://github.com/galaxyproject/galaxy/blob/dev/LICENSE.txt).
- [x] I agree to allow the Galaxy committers to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT). If this condition is an issue, uncheck and just let us know why with an e-mail to galaxy-committers@lists.galaxyproject.org.

### CODE REVIEW

**Review by bgruening — APPROVED** [review]
Great refactoring! Thanks a lot @jmchilton!

File: test/unit/tools/test_watcher.py [hunk 1]
```diff
@@ -6,7 +6,7 @@
```
Comment (nsoranzo) [inline test/unit/tools/test_watcher.py:6]: Move this file to `test/unit/tool_util/` directory?

File: test/unit/tools/test_toolbox_filters.py [hunk 2]
```diff
@@ -1,4 +1,4 @@
-from galaxy.tools.toolbox.filters import FilterFactory
+from galaxy.tool_util.toolbox.filters import FilterFactory
```
Comment (nsoranzo) [inline test/unit/tools/test_toolbox_filters.py:1]: Move this file to `test/unit/tool_util/` directory?

File: test/unit/tools/test_tool_panel.py [hunk 3]
```diff
@@ -1,4 +1,4 @@
-from galaxy.tools.toolbox import ToolSection
+from galaxy.tool_util.toolbox import ToolSection
```
Comment (nsoranzo) [inline test/unit/tools/test_tool_panel.py:1]: Move this file to `test/unit/tool_util/` directory?

File: lib/galaxy/tools/filters/__init__.py [hunk 4]
```diff
@@ -1,5 +1,5 @@
 """Deprecated module for toolbox filters.
 
 Filters placed in this module will still work - but filters should be
-moved to lib/galaxy/tools/toolbox/filters.
```
Comment (nsoranzo) [inline lib/galaxy/tools/filters/__init__.py:4]: Should we have a copy of this file also in `lib/galaxy/tools/toolbox/filters` ?

Comment (jmchilton) [reply lib/galaxy/tools/filters/__init__.py:4]: I was going to wait for someone to complain, but perhaps we should.

**Review by nsoranzo — APPROVED** [review]

**Comment by nsoranzo** [comment]
Ah, a couple of import in unit tests need to be updated.

**Comment by jmchilton** [comment]
And we don't package the unit tests so there is not a very good way to setup those dependencies for the packaged Galaxy variants. Can I just undo those last few changes and keep the unit tests in tools/ until we are properly packaging unit tests.

**Comment by nsoranzo** [comment]
> And we don't package the unit tests so there is not a very good way to setup those dependencies for the packaged Galaxy variants. Can I just undo those last few changes and keep the unit tests in tools/ until we are properly packaging unit tests.

I'm not sure I understand what you mean, but I've pushed a commit that fixes the unit tests (both for "normal" Galaxy and in packages). Does this look OK?

**Review by nsoranzo — APPROVED** [review]

**Comment by jmchilton** [comment]
Thanks @nsoranzo !

**Pull request merged**